applications/vanilla/models/class.discussionmodel.php

```diff
@@ -1009,7 +1009,8 @@ public function calculate(&$discussion) {
         }
 
         // Allow for discussions to be archived
-        if ($discussion->DateLastComment && Gdn_Format::toTimestamp($discussion->DateLastComment) <= $archiveTimestamp) {
+        $dateLastCommentTimestamp = $discussion->DateLastComment ? Gdn_Format::toTimestamp($discussion->DateLastComment) : null;
```

Comment:
I'm not sure you need the ternary operator here ?

Wouldn't

```
$dateLastCommentTimestamp = Gdn_Format::toTimestamp($discussion->DateLastComment);
```

be enough ?